graylog2-web-interface/src/components/common/PaginatedList.tsx

```diff
@@ -52,38 +53,29 @@ const PaginatedList = ({
   hideFirstAndLastPageLinks,
   hidePreviousAndNextPageLinks,
   onChange,
-  pageSize: propsPageSize,
+  pageSize: propPageSize,
   pageSizes,
   showPageSizeSelect,
   totalItems,
+  useQueryParameter,
 }: Props) => {
-  const initialPage = activePage > 0 ? activePage : INITIAL_PAGE;
-  const [pageSize, setPageSize] = useState(propsPageSize);
-  const [currentPage, setCurrentPage] = useState(initialPage);
-  const numberPages = pageSize > 0 ? Math.ceil(totalItems / pageSize) : 0;
+  const { page, setPage, pageSize: queryParamPageSize, setPageSize } = usePaginationQueryParameter(pageSizes);
+  const currentPage = useQueryParameter ? page : Math.max(activePage, INITIAL_PAGE);
+  const pageSize = (useQueryParameter && showPageSizeSelect) ? queryParamPageSize : propPageSize;
 
-  useEffect(() => {
-    if (activePage > 0) {
-      setCurrentPage(activePage);
-    }
-  }, [activePage]);
-
-  useEffect(() => {
-    setPageSize(propsPageSize);
-  }, [propsPageSize]);
+  const numberPages = pageSize > 0 ? Math.ceil(totalItems / pageSize) : 0;
 
   const _onChangePageSize = (event: React.ChangeEvent<HTMLOptionElement>) => {
     event.preventDefault();
     const newPageSize = Number(event.target.value);
 
-    setCurrentPage(INITIAL_PAGE);
-    setPageSize(newPageSize);
-    onChange(INITIAL_PAGE, newPageSize);
+    if (useQueryParameter) setPageSize(newPageSize);
```

Comment:
I think we should avoid single line if statement just to be safe.
```suggestion
    if (useQueryParameter) { 
      setPageSize(newPageSize);
    }
```